operations/deployment/terraform/postgres.tf

```diff
@@ -0,0 +1,263 @@
+resource "aws_security_group" "pg_security_group" {
+  name        = var.security_group_name_pg
+  description = "SG for ${var.aws_resource_identifier} - PG"
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = {
+    Name = "${var.aws_resource_identifier}-pg"
+  }
+}
+
+
+resource "aws_security_group_rule" "ingress_postgres" {
+  type              = "ingress"
+  description       = "${var.aws_resource_identifier} - pgPort"
+  # TODO: parameterize the ports
+  from_port         = 5432
+  to_port           = 5432
+  protocol          = "tcp"
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = aws_security_group.pg_security_group.id
+}
+
+data "aws_vpc" "default" {
+  default = true
+} 
+data "aws_subnets" "vpc_subnets" {
+  filter {
+    name   = "vpc-id"
+
+    # todo: support a specified vpc id
+    # values = [var.vpc_id ? var.vpc_id : data.aws_vpc.default.id]
+    values = [data.aws_vpc.default.id]
+  }
+}
+
+module "rds_cluster" {
+  depends_on     = [data.aws_subnets.vpc_subnets]
+  source         = "terraform-aws-modules/rds-aurora/aws"
+  version        = "v7.6.0"
+  name           = var.aws_resource_identifier
+  engine         = var.postgres_engine
+  engine_version = var.postgres_engine_version
+  instance_class = var.postgres_instance_class
+  instances = {
+    1 = {
+      instance_class = var.postgres_instance_class
+    }
+  }
+
+    # Todo: handle vpc/networking explicitly
+  # vpc_id                 = var.vpc_id
+  # allowed_cidr_blocks    = [var.vpc_cidr]
+  subnets                  = var.postgres_subnets == null || length(var.postgres_subnets) == 0 ? data.aws_subnets.vpc_subnets.ids : var.postgres_subnets
```

Comment:
we need a ticket for better vpc management.  see:
https://github.com/bitovi/github-actions-deploy-stackstorm/blob/main/action.yaml#L50-L55